test/SlowTests/Issues/RavenDB-15772.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations.Configuration;
+using Raven.Client.ServerWide.Operations;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_15772 : RavenTestBase
+    {
+        public RavenDB_15772(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task ShouldntThrowConcurrencyException()
+        {
+            using var store = GetDocumentStore();
+            var record = await store.Maintenance.Server.SendAsync(new GetDatabaseRecordOperation(store.Database));
+            var fixedOrder = record.Topology.AllNodes.ToList();
+
+            var updateClientConfig = store.Maintenance.SendAsync(
+                new PutClientConfigurationOperation(
+                    new ClientConfiguration
+                    {
+                        MaxNumberOfRequestsPerSession = 30
+                    }));
+
+            var updateTopology = store.Maintenance.Server.SendAsync(new ReorderDatabaseMembersOperation(store.Database, fixedOrder, true));
+
+            await Task.WhenAll(updateClientConfig, updateTopology);
```

Comment:
let's read the configuration and assert it was indeed changed

Reply:
Done